File: 00094. Binary Tree Inorder Traversal/00094.c
void inorder(int *arr, int *index, struct TreeNode *root) {
    if (root != NULL) {
        inorder(arr, index, root->left);

        arr[(*index)++] = root->val;

        inorder(arr, index, root->right);
    }
}

int * inorderTraversal(struct TreeNode *root, int *returnSize) {
    *returnSize = 0;
    
    if (root != NULL) {
        int *ret = (int *) malloc(sizeof(int) * 100);
    
        inorder(ret, returnSize, root);
        
        return ret;
    }
    
    return NULL;
}

File: LeetCode/00094. Binary Tree Inorder Traversal/00094.go
// https://leetcode.com/problems/binary-tree-inorder-traversal

func inorderTraversal(root *TreeNode) []int {
    if root == nil {
        return nil
    }

    ans := append(inorderTraversal(root.Left), root.Val)
    ans = append(ans, inorderTraversal(root.Right)...)

    return ans
}
